salt/.pylintrc

```diff
@@ -0,0 +1,704 @@
+suggestion-mode=yes
+
+# Allow loading of arbitrary C extensions. Extensions are imported into the
+# active Python interpreter and may run arbitrary code.
+unsafe-load-any-extension=no
+
+[MESSAGES CONTROL]
+
+# Only show warnings with the listed confidence levels. Leave empty to show
+# all. Valid levels: HIGH, INFERENCE, INFERENCE_FAILURE, UNDEFINED.
+confidence=
+
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once). You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use "--disable=all --enable=classes
+# --disable=W".
+disable=R,
+  locally-disabled,
+  file-ignored,
+  unexpected-special-method-signature,
+  import-error,
+  no-member,
+  unsubscriptable-object,
+  un-indexed-curly-braces-error,
+  whitespace-before-colon,
+  indentation-is-not-a-multiple-of-four-comment,
+  blacklisted-name,
+  invalid-name,
+  missing-docstring,
+  empty-docstring,
+  misplaced-comparison-constant,
+  unidiomatic-typecheck,
+  wrong-import-order,
+  ungrouped-imports,
+  wrong-import-position,
+  bad-mcs-method-argument,
+  bad-mcs-classmethod-argument,
+  line-too-long,
+  too-many-lines,
+  bad-continuation,
+  exec-used,
+  attribute-defined-outside-init,
+  protected-access,
+  reimported,
+  fixme,
+  global-statement,
+  unused-variable,
+  unused-argument,
+  redefined-outer-name,
+  redefined-builtin,
+  undefined-loop-variable,
+  logging-format-interpolation,
+  invalid-format-index,
+  line-too-long,
+  unexpected-indentation-comment,
+  continuation-line-indentation-is-not-a-multiple-of-four,
+  continuation-line-missing-indentation-or-outdented,
+  closing-bracket-does-not-match-indentation-of-opening-brackets-line,
+  closing-bracket-does-not-match-visual-indentation,
+  continuation-line-does-not-distinguish-itself-from-next-logical-line,
+  continuation-line-over-indented-for-hanging-indent,
+  continuation-line-over-indented-for-visual-indent,
+  continuation-line-under-indented-for-visual-indent,
+  visually-indented-line-with-same-indent-as-next-logical-line,
+  unaligned-for-hanging-indent,
+  block-comment-should-start-with-cardinal-space,
+  too-many-leading-hastag-for-block-comment,
+  module-level-import-not-at-top-of-file,
+  do-not-assign-a-lambda-expression-use-a-def,
+  3rd-party-local-module-not-gated,
+  pep8-reserved-keywords,
+  str-format-in-logging,
+  import-outside-toplevel,
+  deprecated-method,
+  repr-flag-used-in-string,
+  keyword-arg-before-vararg,
+  incompatible-py3-code,
+  multiple-spaces-before-operator
```

Comment:
Isn't this a lot to disable? Recommended by Salt? Some checks like `unused-variable`/`undefined-loop-variable` may be useful, don't you think?

Reply:
Yes maybe, btw this one just get copied from pylintrc from salt repo, but indeed we may want to tune it a bit (I remember I already did some minor changes)

Reply:
OK let's keep it like this for now, and add some debt ticket to track that we'll want to include more rules :)

Reply:
I will remove from ignore list (and fix our code if needed) for at least those 
```
whitespace-before-colon,
indentation-is-not-a-multiple-of-four-comment,
whitespace-before-colon,
indentation-is-not-a-multiple-of-four-comment,
undefined-loop-variable,
unexpected-indentation-comment,
continuation-line-indentation-is-not-a-multiple-of-four,
continuation-line-missing-indentation-or-outdented,
closing-bracket-does-not-match-indentation-of-opening-brackets-line,
closing-bracket-does-not-match-visual-indentation,
continuation-line-does-not-distinguish-itself-from-next-logical-line,
continuation-line-over-indented-for-hanging-indent,
continuation-line-over-indented-for-visual-indent,
continuation-line-under-indented-for-visual-indent,
visually-indented-line-with-same-indent-as-next-logical-line,
unaligned-for-hanging-indent,
block-comment-should-start-with-cardinal-space,
too-many-leading-hastag-for-block-comment,
pep8-reserved-keywords,
str-format-in-logging,
```
Tell me if you have others that you think we should not ignore (anyway we could remove then from ignore list  in the future if needed)